src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -71,6 +63,22 @@ protected virtual IServiceProvider ServiceProvider
             }
         }
 
+        /// <summary>
+        /// Updates this instance with a configured service Provider.
+        /// </summary>
+        /// <param name="serviceProvider">A ready to use service provider.</param>
+        public void UpdateContainer(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            _serviceCollection = null;
+            _serviceProvider = serviceProvider;
```

Comment:
You may need to use a lock() around the _serviceProvider since this is definitely not going to be thread safe. 

You are doing about 3 operations there so don't think it can be avoided.

Reply:
Will this suffice?

```c#
public void UpdateContainer(IServiceProvider serviceProvider)
{
    lock (_syncLock)
    {
        if (serviceProvider == null)
        {
            throw new ArgumentNullException(nameof(serviceProvider));
        }

        _serviceCollection = null;
        _isImmutable = true;
        _serviceProvider = serviceProvider;
    }
}
```

Reply:
Will this suffice?

```c#
public void UpdateContainer(IServiceProvider serviceProvider)
{
    if (serviceProvider == null)
    {
        throw new ArgumentNullException(nameof(serviceProvider));
    }

    lock (_syncLock)
    {

        _serviceCollection = null;
        _serviceProvider = serviceProvider;
        _isImmutable = true;
    }
}
```